### PR TITLE
🖼 Set any url as profile picture (useful for NFTs)

### PR DESCRIPTION
**Pull request by shivekkhurana**
Introduces an RPC api on multiaccounts endpoint to add any url as a profile picture.

In case the URL is a square image, it is not cropped. If it's rectangular, it's cropped to get the largest square with centre same as the centre of the rectangle. 

This feature is required for setting NFTs as profile picture.

### CODE REVIEW

**Review by alaibe — APPROVED** [review]

File: images/decode.go [hunk 1]
```diff
@@ -27,6 +30,20 @@ func Decode(fileName string) (image.Image, error) {
 	return decodeImageData(fb, file)
 }
 
+func DecodeFromURL(path string) (image.Image, error) {
+	res, err := http.Get(path)
```
Comment (richard-ramos) [inline images/decode.go]: Consider adding a timeout:
```suggestion
	client := http.Client{
		Timeout: 5 * time.Second,
	}
	res, err := client.Get(path)
```

File: images/decode.go [hunk 2]
```diff
@@ -27,6 +30,20 @@ func Decode(fileName string) (image.Image, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (richard-ramos) [inline images/decode.go:44]:
```suggestion
defer func() {
	if err := res.Body.Close(); err != nil {
		log.Error("failed to close profile pic http request body", "err", err)
	}
}()
```

**Review by richard-ramos — COMMENTED** [review]
Good work!
I only have some minor observations

File: images/decode.go (hunk 1 above)
Comment (shivekkhurana) [reply images/decode.go]: This suggestions feature is cool.

**Review by richard-ramos — APPROVED** [review]

**Review by Samyoul — APPROVED** [review]
Functionally, this work is great. I do have one observation, does the user know that they are accessing a 3rd party service directly and therefore exposing their IP address?

**Comment by shivekkhurana** [comment]
@Samyoul 
> Functionally, this work is great. I do have one observation, does the user know that they are accessing a 3rd party service directly and therefore exposing their IP address?

Before the user is able to view collectibles, they explicitly agree to the fact that their IP will be exposed. This function (given the current ui) will be called only after the user's permission is collected. 

However this function doesn't assume the permission and can be called anytime.

**Comment by Samyoul** [comment]
@shivekkhurana This  what I assumed, we don't really police these settings in `status-go` we just assume that `status-react` is doing it for us. My concern is that this functionality can be used for any remote image source, not necessarily NFTs. Although in the case where a user is explicitly selecting an image from a remote source, I'd assume that they understand that the app will call a remote 3rd party server and expose their IP ... I suppose in that case if we added functionality to select any remote image source that a permission needs to be granted in the settings managed by `status-react`

**Comment by shivekkhurana** [comment]
>  I suppose in that case if we added functionality to select any remote image source that a permission needs to be granted in the settings managed by status-react

If we add a functionality to select *any* image from a URL, we will add this permission to status-react. As of now, the user can *only* select an NFT as PFP by clicking on it (after explicitly enabling it).

**Pull request merged**